src/main/java/bio/terra/common/LockOperation.java

```diff
@@ -0,0 +1,33 @@
+package bio.terra.common;
+
+import java.util.List;
+
+/** Actions which could be taken to modify the locks held on a resource (dataset, snapshot). */
+public enum LockOperation {
+  LockExclusive,
+  LockShared,
+  UnlockExclusive,
+  UnlockShared;
+
+  /**
+   * @return additional guidance for users investigating a failed lock operation, usually as part of
+   *     a failed job / flight.
+   */
+  public List<String> getErrorDetails() {
+    var template =
+        "A failure to obtain %s on a resource likely means that it's already %s by another process.";
+    return switch (this) {
```

Comment:
When defining an enum I try to avoid `switch` or `if (this == X)` since those cases can usually be written in a data-driven way. I think data-driven is preferable in some cases as it's usually easier to reason about and it's clearer that all cases are handled. One way to write this enum in a data driven way is:
```
public enum LockOperation {
  LOCK_EXCLUSIVE(true, "a shared lock", "exclusively locked"),
  LOCK_SHARED(true, "an exclusive lock", "locked"),
  UNLOCK_EXCLUSIVE,
  UNLOCK_SHARED;

  private static final String TEMPLATE = "A failure to obtain %1$s on a resource likely means that it's already %2$s by another process.";
  private final boolean isLock;
  private final String arg1;
  private final String arg2;

  LockOperation(boolean isLock, String arg1, String arg2) {
    this.isLock = isLock;
    this.arg1 = arg1;
    this.arg2 = arg2;
  }

  LockOperation() {
    this(false, null, null);
  }

  /**
   * @return additional guidance for users investigating a failed lock operation, usually as part of
   * a failed job / flight.
   */
  public List<String> getErrorDetails() {
    if (arg1 != null && arg2 != null) {
      return List.of(TEMPLATE.formatted(arg1, arg2));
    }
    return List.of();
  }

  /**
   * @return whether this lock operation represents an attempted lock application (rather than
   * unlock)
   */
  public boolean lockAttempted() {
    return isLock;
  }
}
```

Reply:
You could get rid of `isLock` if you didn't mind conflating its state and the presence of `arg1` and `arg2`.